fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/BlobResourceScanner.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.blob;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.logging.Logger;
+
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.ibm.fhir.persistence.jdbc.dao.api.ResourceRecord;
+
+/**
+ * Scans resource payload records which have been stored in the blob
+ * store. Used by the reconciliation process to identify records
+ * existing in the blob store which do not have a corresponding
+ * record in the RDBMS.
+ */
+public class BlobResourceScanner {
+    private static final Logger logger = Logger.getLogger(BlobResourceScanner.class.getName());
+
+    // Where to start scanning from
+    private String continuationToken;
+
+    // A function used to process a page of blob items
+    private final Consumer<List<ResourceRecord>> pageHandler;
+
+    /**
+     * Public constructor
+     * @param marker the Azure Blob marker token from which to start the scan
+     * @param pageHandler
+     */
+    public BlobResourceScanner(String continuationToken, Consumer<List<ResourceRecord>> pageHandler) {
```

Comment:
java: Some methods in this class have params that do not match the javadoc params.

Reply:
fixed